src/state.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright: ConsensusLab
+//
+use cid::Cid;
+use fil_actors_runtime::Map;
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::Cbor;
+use fvm_ipld_hamt::BytesKey;
+use fvm_shared::address::Address;
+use fvm_shared::MethodNum;
+use ipc_gateway::IPCAddress;
+use primitives::{TCid, THamt};
+use serde::{Deserialize, Serialize};
+use serde_tuple::{Deserialize_tuple, Serialize_tuple};
+use std::collections::HashSet;
+
+use crate::ConstructorParams;
+
+#[derive(Serialize, Deserialize)]
+pub struct State {
+    pub ipc_gateway_address: Address,
+    pub registry: RegistryCid, // (exec_id, actors) -> pre-commitments
+}
+impl Cbor for State {}
+
+type RegistryCid = TCid<THamt<RegistryKey, RegistryEntryCid>>;
+type RegistryEntryCid = TCid<THamt<IPCAddress, MethodNum>>;
```

Comment:
We may be able to simplify this a bit. Having a `Hamt<Hamt>` means that you have to load twice from the blockstore (and there is an overhead to pass the WASM boundary through the syscall). I don't think we'll have a large number of actors involved in an execution, we can probably use a plain old slice or HashMap for entries.

Reply:
Thanks for pointing the out! :+1:

Reply:
Would it also make difference if we use a hash of `RegistryKey` as the key in the registry HAMT?

Reply:
I tried to simplify the registry type in 0726536.

Reply:
> Would it also make difference if we use a hash of RegistryKey as the key in the registry HAMT?

The HAMT handle this for you under the hood. I think the current approach only does a single call to the blockstore and is simpler. Thanks.

Reply:
@adlrocha does it look fine now?

Reply:
I think it would make sense to use a hash/CID as a key instead of plain pair (`RegistryKey`), because HAMT seems to store the plain key, although we don't seem to really need it.